server/src/test/YmlKaoFileVisitor.test.ts

```diff
@@ -0,0 +1,247 @@
+      done();
+    });
+    // tslint:disable-next-line: max-line-length
+    it("should parse a well written YML file containing instances and functions and provide completion for them", (done) => {
+      const inputStream = new ANTLRInputStream(`
+        function functionWithoutArgs()
+          --> domains Text
+          --> action {
+            return "it works";
+          }
+        ;
+
+        Text simpleInstance;
+
+        function functionWithoutArgs2
+          args {}
+          --> domains Text
+          --> action {
+            return "it works";
+          }
+        ;
+
+        Collection collection;
+
+        function functionWithArgsAsBlock
+          args {
+            Object arg1
+            Text arg2
+          }
+          --> domains Text
+          --> return "it works"
+        ;
+
+        Collection collectionWithLevel2
+        --> domainsLevel2 Text
+        ;
+
+        function functionWithoutArgsWithPar(Object arg1, Text arg2)
+          --> domains Text
+          --> return "it works"
+        ;
+      `);
+      const lexer = new YmlToBdlLexer(inputStream);
+      const tokenStream = new CommonTokenStream(lexer);
+      const parser = new YmlToBdlParser(tokenStream);
+
+      const result = parser.kaoFile();
+      const completionItems: CompletionItem[] = [];
+      const visitor = new YmlKaoFileVisitor(completionItems);
+      visitor.visit(result);
+      const expectedCompletionItems = [
+        {
+          data: "id_static_functionWithoutArgs",
+          detail: "Text",
+          documentation: "not documented",
+          kind: 3,
+          label: "functionWithoutArgs",
+        },
+        {
+          data: "id_static_simpleInstance",
+          detail: "Text",
+          documentation: "not documented",
+          kind: 6,
+          label: "simpleInstance",
+        },
+        {
+          data: "id_static_functionWithoutArgs2",
+          detail: "Text",
+          documentation: "not documented",
+          kind: 3,
+          label: "functionWithoutArgs2",
+        },
+        {
+          data: "id_static_collection",
+          detail: "Collection",
+          documentation: "not documented",
+          kind: 6,
+          label: "collection",
+        },
+        {
+          data: "id_static_functionWithArgsAsBlock",
+          detail: "Text",
+          documentation: "not documented",
+          kind: 3,
+          label: "functionWithArgsAsBlock",
+        },
+        {
+          data: "id_static_collectionWithLevel2",
+          detail: "Collection − Text",
+          documentation: "not documented",
+          kind: 6,
+          label: "collectionWithLevel2",
+        },
+        {
+          data: "id_static_functionWithoutArgsWithPar",
+          detail: "Text",
+          documentation: "not documented",
+          kind: 3,
+          label: "functionWithoutArgsWithPar",
+        },
+      ];
+      assert.strictEqual(completionItems.length, 7);
```

Comment:
OK if `deepEqual` provides a crappy message if ≠ in length. Unnecessary if it handles that in a nice way. I think. Dunno. Do as you wish.